midend/interpreter.cpp

```diff
@@ -859,6 +859,17 @@ void ExpressionEvaluator::postorder(const IR::Operation_Unary* expression) {
         return;
     }
     if (sv->isUnknown()) {
+        if (auto cast = expression->to<IR::Cast>()) {
+            if (cast->destType->is<IR::Type_Boolean>() &&
+                (l->is<SymbolicInteger>() || l->is<SymbolicBool>())) {
```

Comment:
Additional conditions are needed in order not to fall into the `else` case when processing replaced enums in cast